Makefile

```
@@ -569,6 +569,14 @@ $(info I CC:        $(shell $(CC)   --version | head -n 1))
 $(info I CXX:       $(shell $(CXX)  --version | head -n 1))
 ifdef LLAMA_CUBLAS
 $(info I NVCC:      $(shell $(NVCC) --version | tail -n 1))
+CUDA_VERSION := $(shell nvcc --version | grep -oP 'release (\K[0-9]+\.[0-9])')
+ifeq ($(shell echo "$(CUDA_VERSION) < 11.7" | bc),1)
```

awk is perhaps a more standard compile-time dependency than bc:
```suggestion
ifeq ($(shell awk -v "v=$(CUDA_VERSION)" 'BEGIN { print (v < 11.7) }'),1)
```
e.g. Arch Linux has awk in base and base-devel, while bc is in neither.

---

Why not like this?

```suggestion
ifeq ($(shell awk 'BEGIN { print ($(CUDA_VERSION) < 11.7) }'),1)
```

To me this is both more intuitive and simpler.

---

That works too, I just like to be in the habit of keeping data and code separate, if only so you don't get a syntax error (or worse) if the value contains unexpected characters that don't get escaped properly.

---

Okay, the point with syntax errors makes sense.